core/modules/views/views.module

```diff
@@ -2188,3 +2188,30 @@ function views_autoload_info() {
     'ViewsSpecialBlock' => 'includes/views.special_block.inc',
   );
 }
+
+/**
+ * Cleans a CSS identifier, taking into account views tokenization.
+ *
+ * @param sring &$class
+ *   A single CSS class, as entered into the Views user interface.
+ * @param object $plugin
+ *   The complete view object for tokenization.
+ * @param string $row_index
+ *   The index indicating the row of the view.
+ *
+ * @return string $class
+ *   A sanitized clean css identifier with tokens replaced.
+ */
+function _views_tokenized_clean_css_identifier(&$class, $plugin, $row_index) {
+  if (!empty($class)) {
+    $tokenized = strip_tags($plugin->tokenize_value($class, $row_index));
```

Comment:
This will convert the entire string (including explicitly typed underscores) if a token is present. We should be converting underscores to hyphens in tokens, but leaving the user-provided input the same.

Reply:
> This will convert the entire string (including explicitly typed underscores) if a token is present. 

Yes, it will convert the entire class, but not any of the other classes that may have also been added at the same.

> We should be converting underscores to hyphens in tokens, but leaving the user-provided input the same.

I'm not sure how I can tell if a class is only partially tokenized. Suggestions?

Reply:
Maybe chop up the string so that only the square brackets, i.e. tokens, get sent to be tokenized and then reassemble. Not the prettiest idea but might work.

Reply:
@jenlampton Here's a working example of chopping up the classes which combine custom text and tokens and then recombining:

```
function _views_tokenized_clean_css_identifier($class, $plugin, $row_index) {
  if (!empty($class)) {
    $subclasses = preg_split('/(\[[^\]]*])/i', $class, -1, PREG_SPLIT_NO_EMPTY | PREG_SPLIT_DELIM_CAPTURE);
    foreach ($subclasses as &$subclass) {
      $tokenized = $plugin->tokenize_value($subclass, $row_index);
      if ($tokenized == $subclass) {
        // No token replacement, all valid CSS characters allowed.
        $subclass = backdrop_clean_css_identifier($tokenized, array());
      }
      else {
        // For BC token replacements use limited CSS characters.
        $subclass = backdrop_clean_css_identifier($tokenized);
      }
      $class = implode('', $subclasses);
    }
    return $class;
  }
}
```